src/charm.py

```diff
@@ -147,10 +157,37 @@ def _on_munge_available(self, event):
             self.unit.status = BlockedStatus("Error restarting munge")
             event.defer()
 
-    def _on_db_unavailable(self, event):
-        self._stored.db_info = {}
-        # TODO tell slurmctld that slurmdbd left?
-        self._check_status()
+    def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
+        """Set database info in StoredState and invoke _write_config_and_restart_slurmdbd.
```

Comment:
Later on we will move from StoredState to peer relation/integration info instead right?

Reply:
Yes, currently I am targeting just getting slurmdbd to work with DPE's MySQL charm to get off of Percona Cluster. Once we start working to decouple the slurmdbd charm from slurm_ops_manager is when I will start making an effort to get off of StoredState.

Reply:
Big thing right now is that StoredState is being used to track the "state of the world." Many of the methods in the charm are tied to evaluating the current values of slots in the StoredState object.